lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,43 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
+            end
           end
 
           def href(value)
             self._href = value
+            nil
           end
 
           def meta(value)
             self._meta = value
+            nil
           end
 
+          def serializable_hash
```

Comment:
Indeed, it is not a perfect name. The rationale was that `serializable_hash` is already used all around in AMS for objects that are "ready to be transformed into JSON", i.e. no further treatment is required (used in `SerializableResource.new(resource).serializable_hash`, PaginationLink.new(stuff).serializable_resource`).